slasher/service/data_update.go

```diff
@@ -37,3 +46,134 @@ func (s *Service) finalisedChangeUpdater() error {
 		}
 	}
 }
+
+func (s *Service) slasherOldAtetstationFeeder() error {
+	if s.beaconClient == nil {
+		return fmt.Errorf("can't feed old attestations to slasher. beacon client has not been started")
+	}
+	ch, err := s.beaconClient.GetChainHead(s.context, &ptypes.Empty{})
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	if ch.FinalizedEpoch < 2 {
+		log.Info("archive node doesnt have historic data for slasher to proccess. finalized epoch: %d", ch.FinalizedEpoch)
+	}
+	log.Infof("Finalized epoch is currently: %d", ch.FinalizedEpoch)
+	errOut := make(chan error)
+	var errorWg sync.WaitGroup
+	e, err := s.slasherDb.GetLatestEpochDetected()
+	if err != nil {
+		log.Error(err)
+		s.Stop()
+		return err
+	}
+	for i := e; i < ch.FinalizedEpoch; i++ {
+		ats, err := s.beaconClient.ListAttestations(s.context, &ethpb.ListAttestationsRequest{
+			QueryFilter: &ethpb.ListAttestationsRequest_TargetEpoch{TargetEpoch: i},
+		})
+		if err != nil {
+			log.Error(err)
```

Comment:
This function could benefit a lot from more specific error messages, such as `log.WithError(err).Errorf("Could not list attestations for epoch %d", i)`

Reply:
fixed